modelbaker/db_factory/layer_uri.py

```diff
@@ -35,7 +35,7 @@ def __init__(self, uri: str) -> None:
         self.provider = None
 
     @abstractmethod
-    def get_data_source_uri(self, record: dict) -> str:
+    def get_data_source_uri(self, record: dict, multigeom: bool) -> str:
```

Comment:
Instead of adding a new parameter here for all `layer_uri` subclasses, we could handle the `multigeom` property as a class member (perhaps even a bit more specific to make it self-explanatory, like `gpkg_multigeom`), just as we do here for PG: https://github.com/opengisch/QgisModelBakerLibrary/blob/647761c04ac189eece3054494e36d6381f39ef0d/modelbaker/generator/generator.py#L107

Reply:
I agree here. But still I'm a little confused. The `layer_uri` here could be a `GpkgLayerUri` but still we assign a `pg_estimated_metadata`. Doesn't this bother it?

Reply:
I don't think so. Even if you set a `pg_estimated_metadata` for a `GpkgLayerUri` instance, it wouldn't be needed/read by the `GpkgLayerUri` class, i.e., it'll be just ignored. Perhaps we take advantage there of the flexibility given by Python (think about stuff like these: https://stackoverflow.com/questions/12569018/why-is-adding-attributes-to-an-already-instantiated-object-allowed).

Reply:
Makes sense. Thanks.